hoot-core/src/main/cpp/hoot/core/io/HootApiDbWriter.cpp

```diff
@@ -62,11 +63,17 @@ HootApiDbWriter::~HootApiDbWriter()
 
 void HootApiDbWriter::_addElementTags(const shared_ptr<const Element> &e, Tags& t)
 {
-  if (e->getCircularError() >= 0.0)
+  if (!t.contains(MetadataTags::HootStatus()))
   {
-    t[MetadataTags::ErrorCircular()] = QString::number(e->getCircularError());
+    if ((_textStatus && t.getNonDebugCount() > 0) || (_includeDebug && _textStatus))
+      t[MetadataTags::HootStatus()] = e->getStatus().toTextStatus();
+    else if (_includeDebug && !_textStatus)
+      t[MetadataTags::HootStatus()] = QString::number(e->getStatus().getEnum());
   }
-  t[MetadataTags::HootStatus()] = QString::number(e->getStatus().getEnum());
+  if (e->hasCircularError() && _includeCircularError)
+    t[MetadataTags::ErrorCircular()] = QString::number(e->getCircularError());
+  if (_includeDebug || _includeIds)
+    t[MetadataTags::HootId()] = QString("%1").arg(e->getId());
```

Comment:
`_addElementTags` is the major change to this issue.  It was very different from the way `OsmXmlWriter::_writeNodes`, `OsmXmlWriter::_writeWays`, and `OsmXmlWriter::_writeReations` all handled the `error:circular` and `hoot:status` flags.  This change caused multiple tests to be updated.